File: Semestrul 3/MAP/lab 2 MAP/model/QueueContainer.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;

public class QueueContainer extends AbstractContainer {

    @Override
    public Task remove() {
        if (!tasks.isEmpty()) {
            return tasks.remove(0);
        }
        else
            return null;
    }

    @Override
    public void add(Task task) {
        tasks.addLast(task);
    }

}

File: Semestrul 3/MAP/lab 10 MAP/model/QueueContainer.cs
namespace seminar10.model;

public class QueueContainer : AbstractContainer
{
    public override Task Remove()
    {
        if (_tasks.Count == 0)
            return null!;
        var task = _tasks[0];
        _tasks.RemoveAt(0);
        return task;
    }

    public override void Add(Task task)
    {
        _tasks.Add(task);
    }
}